### ISSUE TITLE
Exclude from the MagicNumberViolation check keyword arguments inside a constructor.

### ISSUE THREAD

**Issue by vkurilin** [wemake-services/wemake-python-styleguide#2412]
# Rule request

Exclude from the MagicNumberViolation check keyword arguments inside a constructor.

## Thesis

```
# bad
lat = 3.14
lon = 3.14
MyPoint(
    lat=lat, 
    lon=lon,
)


# good
MyPoint(
    lat=3.14, 
    lon=3.14,
)
```

## Reasoning

Function keyword arguments are excluded from this check. And just like in the case of functions, keyword arguments for a constructor are also self-descriptive.

If we allow using magic numbers in constructor keywords, we achieve both consistency with the usage of functions and less boilerplate code.

**Comment by sobolevn** [comment]
Probably, any named arg can count as a name 🤔 
The only downside here is that you can use it twice:
```python
MyPoint(
    lat=3.15, 
    lon=3.14,
)


# ...

MyPoint(
    lat=3.15, 
    lon=3.14,
)
```

**Comment by vkurilin** [comment]
I would argue that if a rule checks for magic constants, in an ideal world we shouldn't rely on it to protect us from constant overuse. Maybe it's a job for another rule :)

**Comment by sobolevn** [comment]
We already have one ;)

But, we don't two usages as "overuse".

**Comment by alythobani** [comment]
Curious why this was closed as not planned? Not a big deal since it's easy to disable the rule, but just started using wemake-python-styleguide in a project and am gonna disable W432 because of this false positive.

**Comment by sobolevn** [comment]
This is not a false positive, consider this case:

```python
FloatValidator(min=23.4, max=87.3)
```

Why `23.4` and why `87.3`? `min` and `max` don't help here. Compare this code with:

```python
MIN_SLA_CONSUMPTION = 23.4
MAX_LAST_SEASON_CONSUMPTION = 87.3
FloatValidator(min=MIN_SLA_CONSUMPTION, max=MAX_LAST_SEASON_CONSUMPTION)
```

**Comment by alythobani** [comment]
Fair example that there could be value there in extracting those constants. Maybe it's more precise to say that this rule *can* lead to false positives, even if in other cases it validly points out where extracting a constant would be cleaner?

Like you mentioned in your comment above a couple years ago, a named arg can count as the constant's name—which I'm guessing is the reason that WPS432 isn't checked on [keyword arguments inside a function](https://wemake-python-styleguide.readthedocs.io/en/latest/pages/usage/violations/best_practices.html#wemake_python_styleguide.violations.best_practices.MagicNumberViolation)? Which I think the class constructor kwargs in the above examples actually do count as.

As another simple example:

```py
class Person(BaseModel):
    name: str
    age: float

Jake = Person(name="Jake", age=28.5)
Aly = Person(name="Aly", age=30.2)
```

It'd feel pretty overkill to extract out an `ALY_CURRENT_AGE` constant (and `JAKE_CURRENT_AGE`) when the context already makes it clear what 30.2 is.

(Or for a more real-world example, e.g. a `ServiceTier` model with a `price_usd_per_month: float` field)

**Comment by alythobani** [comment]
Oh wait, I just found #2634 (and noticed kwarg numbers in function calls are flagged by this rule) and now I'm confused - do the docs just mean kwarg default values within a function signature, when they say this?

> What are the numbers that we exclude from this check? Any numbers that are assigned to a variable, array, dictionary, **or keyword arguments inside a function**.
